c10/core/DispatchKeySet.h

```diff
@@ -172,7 +172,9 @@ class DispatchKeySet final {
             (1ULL
              << (num_backends + static_cast<uint8_t>(toFunctionalityKey(t)) -
                  1)) -
-            1) {}
+            1) {
+    *this = add(DispatchKey::PythonDispatcher);
```

Comment:
Mostly as a call out (and reminder to myself - just spent a while trying to understand this line haha):

I think one consequence of this is that we need to be careful to use the right API's if we want to the PythonDispatcher to trigger properly during redispatches. AKA, this will properly trigger the PythonDispatcher when it's enabled:
```
// FULL_AFTER ctr will implicitly add the PythonDispatcher key to the keyset, so we can call into it
op.redispatchBoxed(ks & DispatchKeySet(DispatchKeySet::FULL_AFTER, my_dispatch_key));
```

but this redispatch will silently ignore the python dispatcher, even if it's enabled:
```
// PythonDispatcher key never gets added to the keyset, so even if we "turned it on", we'll ignore it
op.redispatchBoxed(DispatchKeySet({my_dispatch_key}));
```

I did a quick audit of `redispatch()` call-sites and I don't think there are an existing places where that happens today that are meaningful to tracing.